test/acceptance/graphql_schema_test.go

```diff
@@ -7,310 +7,341 @@ package test
-func TestGettingTypeNames(t *testing.T) {
-	t.Skip()
+const enableNetworkQueryComparison bool = false
+
+/*
+Loop through all branches of the expected schema and compare each leaf to an
+actual schema retrieved from an introspection query on a local weaviate
+
+Note: Can't compare nested lists ([[][]]) properly, but that is out of scope for the current case
+*/
+func TestCompareExpectedToActualSchemaWithIntrospection(t *testing.T) {
 	t.Parallel()
 
-	query := `{ 
-    __schema { 
-      types {
-        name
-      }
-    }
-  }`
+	// get expected schema
+	rawFile, _ := ioutil.ReadFile("../graphql_schema/schema_design.json")
+	var data map[string]interface{}
+	err := json.Unmarshal(rawFile, &data)
 
-	expectedTypes := []string{"WeaviateObj", "WeaviateLocalObj", "WeaviateLocalConvertedFetchFilterInpObj", "FetchFilterANDInpObj", "FetchFilterORInpObj", "FetchFilterEQInpObj", "FetchFilterFieldANDInpObj", "FetchFilterFieldORInpObj", "FetchFilterNEQInpObj", "FetchFilterIEInpObj", "WeaviateLocalConvertedFetchObj", "WeaviateLocalConvertedFetchThingsObj", "WeaviateLocalConvertedFetchActionsObj", "WeaviateLocalMetaFetchFilterInpObj", "WeaviateLocalMetaFetchObj", "WeaviateLocalMetaFetchGenericsObj", "WeaviateLocalMetaFetchGenericsThingsObj", "WeaviateLocalMetaFetchGenericsActionsObj"}
+	if err != nil {
+		fmt.Println(err)
+	} else {
+		fmt.Println("data successfully read")
+	}
+	expectedSchema := data["data"].(map[string]interface{})
 
+	// get actual schema
+	query := genQuery()
 	response := queryGraphqlAssertOK(t, helper.RootAuth, query)
+	actualSchema := response.Result.(map[string]interface{})
 
-	for _, expected := range expectedTypes {
-		found := false
+	//compare expected schema to actual schema, element by element
+	for expectedTopLayerKey, expectedTopLayerValue := range expectedSchema {
+		actualTopLayerValue := actualSchema[expectedTopLayerKey].(map[string]interface{})
+		traverseNestedSchemaLayer(t, expectedTopLayerKey, expectedTopLayerValue.(map[string]interface{}), actualTopLayerValue)
+	}
+}
 
-		// Check if this expected type is in the response
-		for _, type_ := range response.AssertKey(t, "__schema").AssertKey(t, "types").AssertSlice(t) {
-			foundType := type_.AssertKey(t, "name").AssertString(t)
-			if expected == foundType {
-				found = true
-				break
+func traverseNestedSchemaLayer(t *testing.T, expectedLayerKey string, expectedLayerValue map[string]interface{}, actualLayerValue map[string]interface{}) {
+	// attempt to add the name of the current element to the path
+	updatedExpectedLayerKey := expectedLayerKey
+	if expectedLayerName, ok := expectedLayerValue["name"]; ok {
+		updatedExpectedLayerKey = fmt.Sprintf("%s_%s", updatedExpectedLayerKey, expectedLayerName)
+	}
+	// ensure no tests fail due to minute description differences in the graphql core library implementations in Go and JS
+	if !strings.Contains(updatedExpectedLayerKey, "irective") && !strings.Contains(updatedExpectedLayerKey, "ubscription") {
```

Comment:
You're ignoring Directive's and Subscriptions?

Reply:
Yes, Directives and Subscriptions are part of the core library. There are a handful of differences in element descriptions between the Go and JS implementations that raise errors in this test, but these are not of interest to us (see issue 390 in graphql-go for an example).

Reply:
Could you add the content of your comment to the code too?

Reply:
Done